ui/components/src/utils/theme-gen.ts

```diff
@@ -0,0 +1,188 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useTheme } from '@mui/material';
+import merge from 'lodash/merge';
+import { EChartsTheme, PersesChartsTheme } from '../model';
+
+const DEFAULT_TEXT_COLOR = '#222';
+
+export function useGenerateChartsTheme(
+  themeName: string,
+  echartsThemeOverrides: EChartsTheme
+): PersesChartsTheme | undefined {
+  const muiTheme = useTheme();
```

Comment:
This is great. The one thing I might change (and you don't _have_ to do this), is to make the MUI theme an argument to the function, rather than calling `useTheme` here. That way this is just a plain old function and doesn't need to be a hook or nested underneath the MUI theme provider. But not sure if you did this for a reason, so feel free to leave it as-is if you like it better this way.

Reply:
I tried adding a more simple generateChartsTheme function a few different ways, but got different errors with MUI types, [here is one example](https://user-images.githubusercontent.com/9369625/176187127-85e9c8fc-4f5b-4a58-ad53-9777cc63cab6.png) from when I moved ThemeExtension.d.ts like you suggested previously (is ThemeOptions correct type to use?). 

The current useGenerateChartsTheme approach seemed better than having PersesThemeProvider, but I agree isn't the ideal solution. I do think we'll need to heavily refactor our theme (ex: DarkModeContextProvider approach is out of date), so maybe as part of that, I could look more into the generateChartsTheme issues.